subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -161,6 +163,51 @@ static enum bt_gatt_ots_oacp_res_code oacp_delete_proc_validate(
 }
 #endif
 
+#if defined(CONFIG_BT_OTS_OACP_CHECKSUM_SUPPORT)
+static enum bt_gatt_ots_oacp_res_code oacp_checksum_proc_validate(
+	struct bt_conn *conn,
+	struct bt_ots *ots,
+	struct bt_gatt_ots_oacp_proc *proc)
+{
+	struct bt_gatt_ots_oacp_cs_calc_params *params = &proc->cs_calc_params;
+	const uint8_t *obj_data;
+
+	LOG_DBG("Validating Checksum procedure with offset: 0x%08X and "
+		"length: 0x%08X", params->offset, params->len);
+
+	CHECKIF(!ots->cur_obj) {
+		return BT_GATT_OTS_OACP_RES_INV_OBJ;
+	}
+
+	CHECKIF(params->offset > ots->cur_obj->metadata.size.cur) {
+		return BT_GATT_OTS_OACP_RES_INV_PARAM;
+	}
+
+	CHECKIF((params->offset + (uint64_t) params->len) > ots->cur_obj->metadata.size.alloc) {
+		return BT_GATT_OTS_OACP_RES_INV_PARAM;
+	}
+
+	CHECKIF(ots->cur_obj->state.type != BT_GATT_OTS_OBJECT_IDLE_STATE) {
+		return BT_GATT_OTS_OACP_RES_OBJ_LOCKED;
+	}
```

Comment:
`CHECKIF` should not be used to verify data from remote devices, as it can be configured to work as an ASSERT, and we do not want to ASSERT if a remote device sends bad data. 

Suggest to just use `if` here

Reply:
Done.